### PR TITLE
Hardening/1380 http request send return value

### PR DESCRIPTION
**Pull request by kzangeli**
Changed the signature of httpRequestSend and httpRequestSendWithCurl, to return an integer and not a string. The string is now an output-parameter to the function instead of the return value.

Fixes issue #1380

### CODE REVIEW

File: test/functionalTest/cases/0880_timeout_for_forward_and_notifications/forward_that_times_out.test [hunk 1]
```diff
@@ -155,13 +155,14 @@ Date: REGEX(.*)
...
-Content-Length: 94
+Content-Length: 147
 Content-Type: application/json
 Date: REGEX(.*)
 
 {
     "errorCode": {
-        "code": "404", 
+        "code": "404",
+        "details": "invalid context provider response",
```
Comment (fgalan) [inline test/functionalTest/cases/0880_timeout_for_forward_and_notifications/forward_that_times_out.test]: That means that now our errors are more descriptive in this case?

Comment (kzangeli) [inline test/functionalTest/cases/0880_timeout_for_forward_and_notifications/forward_that_times_out.test]: I got this as an extra, for 'filling in more info' for returning the text in httpRequestSend ...
Didn't bother to investigate why, as extra info doesn't really hurt

Comment (fgalan) [inline test/functionalTest/cases/0880_timeout_for_forward_and_notifications/forward_that_times_out.test]: Indeed, I think is an improvement.

NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**